packages/application/src/router.ts

```diff
@@ -142,20 +178,30 @@ class Router implements IRouter {
    * match the `IRouter.ICommandArgs` interface.
    */
   route(url: string): void {
-    const { base } = this;
-    const parsed = URLExt.parse(url.replace(base, ''));
-    const path = parsed.pathname;
-    const search = parsed.search;
-    const rules = this._rules;
-
-    rules.forEach((command, pattern) => {
-      if (path.match(pattern)) {
-        this.commands.execute(command, { path, search });
+    const parsed = URLExt.parse(url.replace(this.base, ''));
+    const args = { path: parsed.pathname, search: parsed.search };
+    const matches: Private.Rule[] = [];
+
+    // Collect all rules that match the URL.
+    this._rules.forEach((rule, pattern) => {
+      if (parsed.pathname.match(pattern)) {
+        matches.push(rule);
       }
     });
+
+    // Order the matching rules by rank and execute them.
+    matches.sort((a, b) => a.rank - b.rank).forEach(rule => {
```

Comment:
Does this trigger only the first matching rule, or all of them? Server side routing definitely only triggers one, so it would be a bit surprising if multiple can get triggered.  If the promises are ignored, that also makes it difficult to ensure ordering between them.

Reply:
This triggers all of them. This is a decision all routers need to make. One idea we've been considering is having the resolution of the promise contain a value that can *stop* the further execution of commands. This requires some thought.

Reply:
ok make sense that we still have to figure that out. For the beta I would probably feel better if we only ran the first one.